### ISSUE TITLE
Instead of reading the C# docs, I'm reading your code

### ISSUE THREAD

**Issue by stuible** [NoahBurkholder/bird-tech#1]
How good will my subsequent code be?

**Comment by NoahBurkholder** [comment]
Bahahaha, it'll be hit or miss.

There's a couple of things I'm not proud of that were just to save time during development. Especially in NeuralNet.cs it's all `public` for no good reason when most of it should be `private` or `protected`. Maybe I'll change that one day when I have a little time.

I'd also probably avoid typing out each archetype's differences as I did in BirdIdentity.cs (`SetPersonality(Archetype a)`), just for readability's sake. Maybe use some sort of minimal I/O system so you could avoid having clutter within the script. Readability is often just as important as efficiency, especially for infrequent operations like loading the archetypes.

And in general, there's just a lot of clutter from the messiness of working in Unity. I could distill the scripts down to just their isolated functionality and it would probably read better.

**Comment by stuible** [comment]
Amazing, thanks for the response. Enums are actually a cool ass language feature I've never experienced before.  I have the absolute joy of leading an asp.net project this spring so there's a lot to learn...

**Comment by NoahBurkholder** [comment]
Yeah! Enums are super great! Used in combination with the `#define` keyword you can make things so much more readable. Enums also work brilliantly with the Unity Engine because they express themselves as text dropdown menus within the UI. Similar implementations are probably easily possible outside Unity too. (For more typical software.)

One thing that annoyed me for a little while was that there wasn't an obvious + efficient way to count the number of items in an enumerator set, but here's a nice solution:

```csharp
System.Enum.GetNames(typeof(MyEnum)).Length
```

This returns the number of items in MyEnum in a bit of a roundabout way, but it works! You can then add a new enumerator without needing to worry about updating any secondary constants.

**Comment by stuible** [comment]
Awesome tip, cheers.  That will actually come in handy when I want to associate enum indexes back with their names which I will need to do for the API I'm building.
